src/root.js

```diff
@@ -94,14 +94,21 @@ Root.prototype.load = function load(filename, options, callback) {
     var sync = callback === SYNC; // undocumented
 
     // Finishes loading by calling the callback (exactly once)
+    var throwing = null;
     function finish(err, root) {
+        // If we get here, and we are throwing then something must have
+        // incorrectly caught the error, so we need to rethrow it.
+        if (throwing)
+            throw err;
```

Comment:
Just to double check, are we sure that `err` is set at this point? or maybe `throw throwing;` since it's where you actually saved the error that was thrown?